src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,774 @@
+
+        Args:
+            x_vec: the points at which the distribution is generated
+        Returns:
+            y_vecs: the value of the distributions at the given x points.
+        '''
+
+        # Init storage to return
+        y_vecs = []
+
+        for j in range(self._num_vectors):
+
+            # Get vector and evaluate corresponding function
+            # at the chosen set of points
+            vector = self._fourier_vectors[:, j]
+            y_vec = [self._make_dist(vector, theta) for theta in x_vec]
+
+            # Store and plot data
+            y_vecs.append(y_vec)
+
+        return y_vecs
+
+    def _Holevo_centers(self, vectors=None):
+        '''
+        Gets the Holevo average phase from a distribution.
+        Args:
+            vectors: vectors to calculate the average of. If None,
+                defaults to the vectors stored in the class.
+        Returns:
+            centers: the corresponding average angle of the
+                distribution.
+
+        '''
+        if vectors is None:
+            vectors = self._fourier_vectors
+
+        centers = np.angle(vectors[2, :] + 1j*vectors[1, :])
+        return centers
+
+    def _Holevo_variances(self, vectors=None, maxvar=4*np.pi**2/12):
+        '''
+        Gets the Holevo variance from a distribution
+
+        Args:
+            vectors: Fourier vectors to calculate the variance of.
+                If none, defaults to the vectors stored in the class.
+            maxvar: The maximum allowed variance. Defaults to the
+                maximum of a uniform distribution on [-pi,pi].
+        Returns:
+            variances: The Holevo variances of each vector.
+        '''
+        variances = []
+
+        if vectors is None:
+            vectors = self._fourier_vectors
+
+        for j in range(self._num_vectors):
+            if (vectors[2, j]**2 + vectors[1, j]**2) == 0:
+
+                variances.append(maxvar)
+
+            else:
+                variances.append(min(
+                    maxvar,
+                    4 / (vectors[2, j]**2+vectors[1, j]**2) - 1))
+        return variances
+
+    def _make_matrices(self):
+        ''' Function to make the matrices that define multiplication
+        by sin(theta) and cos(theta). We store these as sparse matrices
+        in CSR format, and initialize them in (row,col,data) format.
+        We assume here that self.max_n is less than self.num_freqs.
+        '''
+
+        # self._matrices[n][a] contains matrix M^a(n+1)
+        # following our paper
+        self._matrices = []
+
+        for n in range(1, self._max_n+1):
+
+            # We generate the matrices ourselves in
+            # COO format, and then create a
+            # CSR matrix directly
+
+            cos_matrix, sin_matrix = self._make_matrix(n)
+
+            # Append to list
+            self._matrices.append([cos_matrix, sin_matrix])
+
+    def _make_matrix(self, n):
```

Comment:
can you give some more explanation of what this is doing? Perhaps refer to some equations from your paper or something? Its pretty difficult to tell.

Reply:
I've added an explicit reference to where this can be found in the paper (and hopefully made the docstring a bit clearer)